xblockutils/helpers.py

```diff
@@ -0,0 +1,17 @@
+"""
+Useful helper methods
+"""
+
+
+def child_isinstance(block, child_id, block_class_or_mixin):
```

Comment:
You're right that it does seem oddly specific, but I feel that this exact need is not uncommon and the current definition of this helper fits flexibly into many different use cases. I worry that any attempt to make it more general would make it slower or more complex, which would defeat the purpose. For example, in all the uses I've had for it so far (in mentoring), I don't want a tree - only immediate children.

You are right that moving it into the runtime would create more avenues for optimization, but I think we need to keep the number of methods that runtimes are expected to implement to a minimum, as there are already at least several different runtimes a typical new XBlock may need to work with (workbench, LMS, Studio preview, split DescriptorSystem, old mongo DescriptorSystem).

All that being said, if we feel like this isn't going to be used much I could move it back into `mentoring` for the time being.